src/export.rs

```diff
@@ -0,0 +1,235 @@
+//! Export support for mmj2 proof files.
+
+use diag::Diagnostic;
+use nameck::Nameset;
+use parser::StatementRef;
+use parser::StatementType;
+use parser::TokenRef;
+use proof::ProofTreeArray;
+use regex::Regex;
+use scopeck::ScopeResult;
+use segment_set::SegmentSet;
+use std::cmp::Ord;
+use std::cmp::Ordering;
+use std::cmp::PartialOrd;
+use std::collections::BinaryHeap;
+use std::error;
+use std::fmt;
+use std::io;
+use std::io::Write;
+use std::str;
+use std::u16;
+
+/// The error type for export::export_mmp().
```

Comment:
If you are interested in error de-boilerplating macros, I found https://github.com/tailhook/quick-error which will help out in this regard.